internal/plugin/external_plugins.go

```diff
@@ -0,0 +1,242 @@
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	go_plugin "plugin"
+	"strings"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/http"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+// ExternalPluginLookupFunc returns all available external plugins.
+type ExternalPluginLookupFunc func(
+	pluginDir string,
+	checksumfile string,
+	logger log.Logger,
+) (map[string]*go_plugin.Plugin, error)
+
+// LoadPluginsFromDir loads all plugins from a specified directory and returns
+// Plugins struct with tcp and http connectors.
+func LoadPluginsFromDir(
+	pluginDir string,
+	checksumsFile string,
+	logger log.Logger,
+) (map[string]*go_plugin.Plugin, error) {
+
+	filePaths, err := getCheckedPlugins(pluginDir, checksumsFile, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return loadPluginFiles(filePaths, logger)
+}
+
+func getCheckedPlugins(
+	pluginDir string,
+	checksumsFile string,
+	logger log.Logger,
+) ([]string, error) {
+
+	files, err := ioutil.ReadDir(pluginDir)
+	if err != nil {
+		return nil, err
+	}
+
+	if checksumsFile == "" {
+		logger.Warnln("Plugin hashes were not provided - tampering will not be detectable!")
+		return getFilePaths(pluginDir, files), nil
+	}
+
+	// We override file listing if we did a verification to prevent additions
+	// to plugins between verification and loading the plugins.
+	if files, err = VerifyPluginChecksums(pluginDir, checksumsFile); err != nil {
+		logger.Errorln(err)
+		return nil, err
+	}
+
+	return getFilePaths(pluginDir, files), nil
+}
+
+func getFilePaths(pluginDir string, files []os.FileInfo) []string {
+	filePaths := []string{}
+	for _, file := range files {
+		filePaths = append(filePaths, path.Join(pluginDir, file.Name()))
+	}
+
+	return filePaths
+}
```

Comment:
Naming nit, though one I actually care about.  I want to get away from using `get`/`Get` for pure functions, such as this one.  I think adding a `get` or `fetch` prefix is ok, and sometime preferable, when we're interacting with the file system or network (eg, `getCheckedPlugins` above), but when we're just returning values with no possibility of error I'd much prefer to stick with "name the function after the thing it returns".